x-pack/plugins/security_solution/server/endpoint/services/actions/clients/crowdstrike/crowdstrike_actions_client.ts

```diff
@@ -0,0 +1,318 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { ActionTypeExecutorResult } from '@kbn/actions-plugin/common';
+import {
+  SUB_ACTION,
+  CROWDSTRIKE_CONNECTOR_ID,
+} from '@kbn/stack-connectors-plugin/common/crowdstrike/constants';
+import type { CrowdstrikeActionRequestCommonMeta } from '../../../../../../common/endpoint/types/crowdstrike';
+import type {
+  CommonResponseActionMethodOptions,
+  ProcessPendingActionsMethodOptions,
+} from '../../..';
+import type { ResponseActionAgentType } from '../../../../../../common/endpoint/service/response_actions/constants';
+import { stringify } from '../../../../utils/stringify';
+import { ResponseActionsClientError } from '../errors';
+import type {
+  ActionDetails,
+  EndpointActionDataParameterTypes,
+  EndpointActionResponseDataOutput,
+  LogsEndpointAction,
+} from '../../../../../../common/endpoint/types';
+import type { IsolationRouteRequestBody } from '../../../../../../common/api/endpoint';
+import type {
+  ResponseActionsClientOptions,
+  ResponseActionsClientWriteActionRequestToEndpointIndexOptions,
+  ResponseActionsClientValidateRequestResponse,
+} from '../lib/base_response_actions_client';
+import { ResponseActionsClientImpl } from '../lib/base_response_actions_client';
+import type { NormalizedExternalConnectorClient } from '../lib/normalized_external_connector_client';
+
+export type CrowdstrikeActionsClientOptions = ResponseActionsClientOptions & {
+  connectorActions: NormalizedExternalConnectorClient;
+};
+
+export class CrowdstrikeActionsClient extends ResponseActionsClientImpl {
+  protected readonly agentType: ResponseActionAgentType = 'crowdstrike';
+  private readonly connectorActionsClient: NormalizedExternalConnectorClient;
+
+  constructor({ connectorActions, ...options }: CrowdstrikeActionsClientOptions) {
+    super(options);
+    this.connectorActionsClient = connectorActions;
+    connectorActions.setup(CROWDSTRIKE_CONNECTOR_ID);
+  }
+
+  protected async writeActionRequestToEndpointIndex<
+    TParameters extends EndpointActionDataParameterTypes = EndpointActionDataParameterTypes,
+    TOutputContent extends EndpointActionResponseDataOutput = EndpointActionResponseDataOutput,
+    TMeta extends {} = {}
+  >(
+    actionRequest: ResponseActionsClientWriteActionRequestToEndpointIndexOptions<
+      TParameters,
+      TOutputContent,
+      TMeta
+    >
+  ): Promise<
+    LogsEndpointAction<TParameters, TOutputContent, TMeta & CrowdstrikeActionRequestCommonMeta>
+  > {
+    const agentId = actionRequest.endpoint_ids[0];
+    const eventDetails = await super.getEventDetailsById<{
+      crowdstrike: { event: { HostName: string } };
+    }>({
+      index: ['logs-crowdstrike.fdr*', 'logs-crowdstrike.falcon*'],
```

Comment:
That's good question, sorry I probably should have added a comment to explain this, will do now.

Crowdstrike integration saves to multiple indexes eg. `logs-crowdstrike.fdr-default` or `logs.crowdstrike.falcon-default` , and even possibly to `logs.crowdstrike.alert`. However I imagine that these 2 are crucial for us to search in. 
I added an `asterisk` because because it doesn't throw an error of 'not found index' which happened when I searched eg. in `logs-crowdstrike.falcon-default` but the index was not created. But also I don't think how much more specific I could be with this index, I guess we should not pass `default` because there actually may be a different `space` used, right? 

Please tell me what you think about the above :) Thank you!